sources/packages/backend/apps/api/src/route-controllers/student-scholastic-standings/_tests_/e2e/student-scholastic-standings.institutions.controller.saveScholasticStanding.e2e-spec.ts

```diff
@@ -707,6 +707,101 @@ describe("StudentScholasticStandingsInstitutionsController(e2e)-saveScholasticSt
+      numberOfUnsuccessfulWeeks: 50,
+    });
+    const application = await saveFakeApplication(
+      db.dataSource,
+      {
+        institutionLocation: collegeFLocation,
+      },
+      {
+        offeringIntensity: OfferingIntensity.partTime,
+        applicationStatus: ApplicationStatus.Completed,
+      },
+    );
+    // Institution token.
+    const institutionUserToken = await getInstitutionToken(
+      InstitutionTokenTypes.CollegeFUser,
+    );
+    const endpoint = `/institutions/scholastic-standing/location/${collegeFLocation.id}/application/${application.id}`;
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .post(endpoint)
+      .send(payload)
+      .auth(institutionUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.UNPROCESSABLE_ENTITY)
+      .expect({
+        message:
+          "Number of unsuccessful weeks cannot exceed the number of offering weeks.",
+        errorType: "INVALID_UNSUCCESSFUL_COMPLETION_WEEKS",
+      });
+  });
+
+  it("Should create new scholastic standing 'Student did not complete program' for a part-time application when number of unsuccessful completion weeks is less than the total number of offering weeks.", async () => {
+    // Arrange
+    const application = await saveFakeApplication(
+      db.dataSource,
+      {
+        institutionLocation: collegeFLocation,
+      },
+      {
+        offeringIntensity: OfferingIntensity.partTime,
+        applicationStatus: ApplicationStatus.Completed,
+      },
+    );
+    // Institution token.
+    const institutionUserToken = await getInstitutionToken(
+      InstitutionTokenTypes.CollegeFUser,
+    );
+
+    // Mock with a valid number of unsuccessful weeks (smaller than total offering weeks).
+    mockFormioDryRun({
+      studentScholasticStandingChangeType:
+        StudentScholasticStandingChangeType.StudentDidNotCompleteProgram,
+      numberOfUnsuccessfulWeeks: 5,
+    });
+
+    const endpoint = `/institutions/scholastic-standing/location/${collegeFLocation.id}/application/${application.id}`;
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .post(endpoint)
+      .send(payload)
+      .auth(institutionUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.CREATED)
+      .expect((response) => {
+        expect(response.body.id).toBeGreaterThan(0);
+      });
+    const queryApplication = await db.application.findOne({
+      select: {
+        id: true,
+        currentAssessment: {
+          id: true,
+          triggerType: true,
+          studentScholasticStanding: { id: true },
+        },
+      },
+      relations: {
+        currentAssessment: {
+          studentScholasticStanding: true,
+        },
+      },
+      where: { id: application.id },
+    });
+    expect(queryApplication.currentAssessment.id).toBe(
+      application.currentAssessment.id,
+    );
+    expect(queryApplication.currentAssessment.triggerType).toBe(
+      AssessmentTriggerType.OriginalAssessment,
+    );
```

Comment:
I did not understand the meaning of these assertions.
Can we please validate something that has been changed by the scholastic update?
```ts
expect(queryApplication.currentAssessment.studentScholasticStanding.unsuccessfulWeeks).toBe(5);
```